sdk/daml-script/test/daml/upgrades/DataChanges.daml

```diff
@@ -286,3 +317,336 @@ templateEnumDowngradeFromOld =
+        with
+          k : OptionalFieldsAddedKey
+          ctl : Party
+        controller ctl
+        do
+          exerciseByKey @OptionalFieldsAdded k (OptionalFieldsAddedCallWithCtl ctl)
+          pure ()
+-}
+
+{- PACKAGE
+name: data-changes-dummy
+versions: 1
+depends: |
+  data-changes-1.0.0
+  data-changes-2.0.0
+-}
+
+{- MODULE
+package: data-changes-dummy
+contents: |
+  module DataChangesDummy where
+
+  import qualified V1.OptionalFieldsAdded as V1
+  import qualified V2.OptionalFieldsAdded as V2
+
+  template TLocalTestDummy
+    with
+      party : Party
+    where
+      signatory party
+      choice TLocalTestDummyChoice : ()
+        controller party
+        do
+          cid <- create
+            (V2.OptionalFieldsAdded
+              party
+              (Some "V1")
+            )
+          let cidV1 = coerceContractId @V2.OptionalFieldsAdded @V1.OptionalFieldsAdded cid
+          exercise cidV1 (V1.OptionalFieldsAddedCallWithCtl party)
+          pure ()
+
+      choice TLocalTestDummyChoiceByKey : ()
+        controller party
+        do
+          cid <- create
+            (V2.OptionalFieldsAdded
+              party
+              (Some "V1")
+            )
+          let cidV1 = coerceContractId @V2.OptionalFieldsAdded @V1.OptionalFieldsAdded cid
+          exerciseByKey @V1.OptionalFieldsAdded (V1.OptionalFieldsAddedKey party) (V1.OptionalFieldsAddedCallWithCtl party)
+          pure ()
+-}
+
+tofswdViaCIDExerciseInCommandGlobal : (Text, Test)
+tofswdViaCIDExerciseInCommandGlobal = templateOptionalFieldsSetWhenDowngradingTest "static,  global,    in command, cid" $ \a b -> do
+  cid <- a `submit` createCmd V2.OptionalFieldsAdded with party = a, newField = Some "V1"
+
+  let cidV1 = coerceContractId @V2.OptionalFieldsAdded @V1.OptionalFieldsAdded cid
+
+  -- Ensure we can only call the V1 choice
+  a `trySubmit` exerciseExactCmd cidV1 V1.OptionalFieldsAddedCall
+
+{- MODULE
+package: data-changes
+contents: |
+  module OptionalFieldOnChoiceAdded where
+
+  template OptionalFieldOnChoiceAdded
+    with
+      party : Party
+    where
+      signatory party
+      choice OptionalFieldOnChoiceAddedCall : Text
+        with
+          existingField : Text
+          newField : Optional Text                 -- @V  2
+        controller party
+        do pure "V1"                           -- @V 1
+        do pure "V2"                           -- @V  2
+-}
+
+templateOptionalFieldOnChoiceSetWhenDowngrading : Test
+templateOptionalFieldOnChoiceSetWhenDowngrading = test $ do
+  a <- allocatePartyOn "alice" participant0
+  cid <- a `submit` createExactCmd V1.OptionalFieldOnChoiceAdded with party = a
+
+  let cidV2 = coerceContractId @V1.OptionalFieldOnChoiceAdded @V2.OptionalFieldOnChoiceAdded cid
+
+  -- Ensure we can only call the V1 choice
+  (actAs a <> packagePreference [dataChangesV1]) `trySubmit` exerciseCmd cidV2 V2.OptionalFieldOnChoiceAddedCall { existingField = "V2", newField = Some "V2" }
```

Comment:
I think this line is redundant?

Reply:
Ah yes, I noticed this while porting to 2.x, deleted there, deleted here now :)